File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEditDeleteVolunteerEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.api.organisation.VolunteerDto;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditDeleteVolunteerEventDto extends OrganizationEventDto {
    private VolunteerDto volunteer;

    OrganizationEditDeleteVolunteerEventDto() {
    }

    public OrganizationEditDeleteVolunteerEventDto(OrganisationId organisationId, VolunteerDto volunteer) {
        super(organisationId);
        this.volunteer = volunteer;
    }

    public VolunteerDto getVolunteer() {
        return volunteer;
    }
}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditDeleteVolunteerEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event;

import de.helfenkannjeder.helfomat.api.organization.VolunteerDto;
import de.helfenkannjeder.helfomat.core.organization.OrganizationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditDeleteVolunteerEventDto extends OrganizationEventDto {
    private VolunteerDto volunteer;

    OrganizationEditDeleteVolunteerEventDto() {
    }

    public OrganizationEditDeleteVolunteerEventDto(OrganizationId organizationId, VolunteerDto volunteer) {
        super(organizationId);
        this.volunteer = volunteer;
    }

    public VolunteerDto getVolunteer() {
        return volunteer;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }

}
